### PR TITLE
Add new DateUtils methods

### PR DESCRIPTION
**Pull request by lfortunier**

### CODE REVIEW

File: src/Utils/DateUtils.php [hunk 1]
```diff
@@ -211,4 +211,114 @@ public static function getLastDayOfMonthYear(int $month, int $year): \DateTime
...
+     * <?php
+     * getMonthsBetweenDates(new \DateTime('2015-05-14'), new \DateTime('2015-09-02'));
+     * ?>
+     * </pre>
+     * The above example will output:
+     * <pre>
+     * ['2015-05', '2015-06', '2015-07', '2015-08', '2015-09']
+     * </pre>
+     */
+    public static function getMonthsBetweenDates(\DateTime $start, \DateTime $end): array
```
Comment (mathieu-ducrot) [inline src/Utils/DateUtils.php:247]: getMonthsBetweenDates => getMonthsBetweenDateTimes

File: src/Utils/DateUtils.php [hunk 2]
```diff
@@ -211,4 +211,114 @@ public static function getLastDayOfMonthYear(int $month, int $year): \DateTime
     {
         return self::getFirstDayOfMonthYear($month, $year)->modify('last day of this month')->setTime(23, 59, 59);
     }
+
+    public static function getFirstDayYearFromDatetime(?\DateTime $datetime = null): \DateTime
```
Comment (mathieu-ducrot) [inline src/Utils/DateUtils.php:215]: Datetime should be writting a capital T like the object (do this for every function that contains DateTime in it)

File: src/Utils/DateUtils.php [hunk 3]
```diff
@@ -211,4 +211,114 @@ public static function getLastDayOfMonthYear(int $month, int $year): \DateTime
...
+     * <?php
+     * getNbOfWorkingDaysBetweenDatetime(new \DateTime('2022-07-18 00:00:00'), new \DateTime('2022-07-22 23:59:59'));
+     * ?>
+     * </pre>
+     * The above example will output:
+     * <pre>
+     * 5
+     * </pre>
+     */
+    public static function getNbOfWorkingDaysBetweenDatetime(\DateTime $dateFrom, \DateTime $dateTo): int
```
Comment (mathieu-ducrot) [inline src/Utils/DateUtils.php:306]: For convention let's always write it with an s when there is multiple dates

getNbOfWorkingDaysBetweenDatetime => getNbOfWorkingDaysBetweenDatetimes

Also rename the params to be $start and $end

File: src/Utils/DateUtils.php [hunk 4]
```diff
@@ -211,4 +238,331 @@ public static function getLastDayOfMonthYear(int $month, int $year): \DateTime
...
+        $dateTime = new \DateTime($timezone);
+        $currentHour = intval($dateTime->format('H'));
+
+        return $currentHour >= 18;
+    }
+
+    /**
+     * Return last time of last day of month of datetime
+     */
+    public static function getLastDayMonth(\DateTime $datetime): \DateTime
```
Comment (mathieu-ducrot) [inline src/Utils/DateUtils.php:395]: @lfortunier from what i see this is the same as getLastDayMonthFromDateTime so i think we don't need this one.

File: src/Utils/DateUtils.php [hunk 5]
```diff
@@ -211,4 +238,331 @@ public static function getLastDayOfMonthYear(int $month, int $year): \DateTime
...
+    {
+        $clonedDateTime = clone $datetime;
+        return $clonedDateTime->modify('last day of this month')->setTime(23, 59, 59);
+    }
+
+    /**
+     * Return last time of last day of last month of datetime
+     */
+    public static function getLastDayPreviousMonth(\DateTime $datetime): \DateTime
```
Comment (mathieu-ducrot) [inline src/Utils/DateUtils.php:404]: @lfortunier to rename getLastDayPreviousMonthFromDateTime to be on part with getLastDayMonthFromDateTime

File: src/Utils/DateUtils.php [hunk 6]
```diff
@@ -211,4 +238,331 @@ public static function getLastDayOfMonthYear(int $month, int $year): \DateTime
...
+    {
+        $clonedDateTime = clone $datetime;
+        return $clonedDateTime->modify('last day of previous month')->setTime(23, 59, 59);
+    }
+
+    public static function getFirstDayNextMonth(\DateTime $datetime): \DateTime
```
Comment (mathieu-ducrot) [inline src/Utils/DateUtils.php:410]: getFirstDayNextMonthFromDateTime

File: src/Utils/DateUtils.php [hunk 7]
```diff
@@ -211,4 +238,331 @@ public static function getLastDayOfMonthYear(int $month, int $year): \DateTime
...
+    {
+        $clonedDateTime = clone $datetime;
+        return $clonedDateTime->modify('first day of next month');
+    }
+
+    /**
+     * Return first time of first day of month of datetime
+     */
+    public static function getFirstDayMonth(\DateTime $datetime): \DateTime
```
Comment (mathieu-ducrot) [inline src/Utils/DateUtils.php:419]: @lfortunier isn't it the same as getFirstDayYearFromDateTime ? if yes remove this one

File: src/Utils/DateUtils.php [hunk 8]
```diff
@@ -211,4 +238,331 @@ public static function getLastDayOfMonthYear(int $month, int $year): \DateTime
...
+     * ?>
+     * </pre>
+     * The above example will output:
+     * <pre>
+     * Janvier
+     * </pre>
+     */
+    public static function getFormattedLongMonth(\DateTime $date): string
+    {
+        $dateFormatter = new \IntlDateFormatter('fr_FR', \IntlDateFormatter::NONE, \IntlDateFormatter::NONE);
```
Comment (mathieu-ducrot) [inline src/Utils/DateUtils.php:461]: pass the $locale as a param from getFormattedLongMonth (with default 'fr_FR')

File: src/Utils/DateUtils.php [hunk 9]
```diff
@@ -211,4 +238,331 @@ public static function getLastDayOfMonthYear(int $month, int $year): \DateTime
...
+     * ?>
+     * </pre>
+     * The above example will output:
+     * <pre>
+     * Janvier 2022
+     * </pre>
+     */
+    public static function getFormattedLongMonthYears(\DateTime $date): string
+    {
+        $dateFormatter = new \IntlDateFormatter('fr_FR', \IntlDateFormatter::NONE, \IntlDateFormatter::NONE);
```
Comment (mathieu-ducrot) [inline src/Utils/DateUtils.php:482]: same here, pass the locale as a function param

File: src/Utils/DateUtils.php (hunk 7 above)
Comment (lfortunier) [reply src/Utils/DateUtils.php:419]: @mathieu-ducrot no you said `getFirstDayYearFromDateTime` -> it's first day of **year** while `getFirstDayMonth` is the first day of **month**

**Pull request merged**